File: services/block-e-chunking/fixtures/code/go/math_utils.go
package main

import (
	"fmt"
	"math"
	"sort"
)

type MathUtils struct{}

func NewMathUtils() *MathUtils {
	return &MathUtils{}
}

func (m *MathUtils) Mean(values []float64) float64 {
	if len(values) == 0 {
		return 0.0
	}
	sum := 0.0
	for _, v := range values {
		sum += v
	}
	return sum / float64(len(values))
}

func (m *MathUtils) Median(values []float64) float64 {
	if len(values) == 0 {
		return 0.0
	}
	sorted := make([]float64, len(values))
	copy(sorted, values)
	sort.Float64s(sorted)
	n := len(sorted)
	if n%2 == 0 {
		return (sorted[n/2-1] + sorted[n/2]) / 2.0
	}
	return sorted[n/2]
}

func (m *MathUtils) StdDev(values []float64) float64 {
	if len(values) == 0 {
		return 0.0
	}
	mean := m.Mean(values)
	sum := 0.0
	for _, v := range values {
		diff := v - mean
		sum += diff * diff
	}
	variance := sum / float64(len(values))
	return math.Sqrt(variance)
}

func (m *MathUtils) Percentile(values []float64, p float64) float64 {
	if len(values) == 0 {
		return 0.0
	}
	sorted := make([]float64, len(values))
	copy(sorted, values)
	sort.Float64s(sorted)
	n := len(sorted)
	k := float64(n-1) * p / 100.0
	f := math.Floor(k)
	c := math.Ceil(k)
	if f == c {
		return sorted[int(k)]
	}
	d0 := sorted[int(f)] * (c - k)
	d1 := sorted[int(c)] * (k - f)
	return d0 + d1
}

func main() {
	utils := NewMathUtils()
	values := []float64{1.0, 2.0, 3.0, 4.0, 5.0}
	fmt.Printf("Mean: %.2f\n", utils.Mean(values))
	fmt.Printf("Median: %.2f\n", utils.Median(values))
	fmt.Printf("StdDev: %.2f\n", utils.StdDev(values))
	fmt.Printf("Percentile 50: %.2f\n", utils.Percentile(values, 50))
}

File: fixtures/code_corpus/python/math_utils.py
import math
from typing import List, Tuple

class MathUtils:
    """Mathematical utilities."""
    
    @staticmethod
    def mean(values: List[float]) -> float:
        """Calculate mean of values."""
        if not values:
            return 0.0
        return sum(values) / len(values)
    
    @staticmethod
    def median(values: List[float]) -> float:
        """Calculate median of values."""
        if not values:
            return 0.0
        sorted_values = sorted(values)
        n = len(sorted_values)
        if n % 2 == 0:
            return (sorted_values[n//2 - 1] + sorted_values[n//2]) / 2
        return sorted_values[n//2]
    
    @staticmethod
    def std_dev(values: List[float]) -> float:
        """Calculate standard deviation."""
        if not values:
            return 0.0
        mean = MathUtils.mean(values)
        variance = sum((x - mean) ** 2 for x in values) / len(values)
        return math.sqrt(variance)
    
    @staticmethod
    def percentile(values: List[float], p: float) -> float:
        """Calculate percentile."""
        if not values:
            return 0.0
        sorted_values = sorted(values)
        k = (len(sorted_values) - 1) * p / 100
        f = math.floor(k)
        c = math.ceil(k)
        if f == c:
            return sorted_values[int(k)]
        d0 = sorted_values[int(f)] * (c - k)
        d1 = sorted_values[int(c)] * (k - f)
        return d0 + d1
